include/effects/matrix/PatternWeather.h

```diff
@@ -165,75 +238,113 @@ class PatternWeather : public LEDStripEffect
     {
         HTTPClient http;
         String url = "http://api.openweathermap.org/data/2.5/forecast"
-            "?lat=" + strLatitude + "&lon=" + strLongitude + "&appid=" + urlEncode(g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey());
+            "?lat=" + strLatitude + "&lon=" + strLongitude + "&cnt=18&appid=" + urlEncode(g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey());
+        debugI("tomorrowURL: %s", url.c_str());
         http.begin(url);
         int httpResponseCode = http.GET();
 
         if (httpResponseCode > 0)
         {
-            AllocatedJsonDocument doc(4096);
-            deserializeJson(doc, http.getString());
+            AllocatedJsonDocument doc(9216);
+            String tomorrowJson = http.getString();
+            DeserializationError de = deserializeJson(doc, tomorrowJson);
+            String count = doc["cnt"];
+            debugI("Return count %s DE code %s", count.c_str(), de.c_str());
             JsonArray list = doc["list"];
 
-            // Get tomorrow's date
-            time_t tomorrow = time(nullptr) + 86400;
-            tm* tomorrowTime = localtime(&tomorrow);
-            char dateStr[11];
-            strftime(dateStr, sizeof(dateStr), "%Y-%m-%d", tomorrowTime);
-
-            iconTomorrow = -1;
-
-            // Look for the temperature data for tomorrow
-            for (size_t i = 0; i < list.size(); ++i)
+            // Get tomorrow's date 
+            time_t tomorrow = time(nullptr) + 86400; 
+            tm* tomorrowTime = localtime(&tomorrow); 
+            char dateStr[11]; 
+            strftime(dateStr, sizeof(dateStr), "%Y-%m-%d", tomorrowTime); 
+
+            float localMin = 999.0;
+            float localMax = 0.0;
+ 
+            // Look for the temperature data for tomorrow  - needs more work
+            int slot = 0;
+            for (size_t i = 0; i < list.size(); i++) 
             {
                 JsonObject entry = list[i];
-                String dt_txt = entry["dt_txt"];
-                if (dt_txt.startsWith(dateStr))
-                {
-                    //Serial.printf("Weather: Updating Forecast: %s", response.c_str());
-                    JsonObject main = entry["main"];
-                    if (main["temp_max"] > 0)
-                        highTemp        = KelvinToLocal(main["temp_max"]);
-                    if (main["temp_min"] > 0)
-                        lowTemp         = KelvinToLocal(main["temp_min"]);
 
-                    String iconIdTomorrow = entry["weather"][0]["icon"];
-                    iconTomorrow = iconIdTomorrow.toInt();
+                // convert the entry UTC to localtime
+                // if it is tomorrow then figure out the min and max and get the icon
+                time_t entry_time = entry["dt"];
+                tm* entryLocal = localtime(&entry_time);
+                char entryStr[11];
+                strftime(entryStr, sizeof(entryStr), "%Y-%m-%d", entryLocal);

```

Comment:
I really need a refresher on the standard class libraries  That is very nice.

This is why we do this so we can learn new things.

Reply:
If I came across as objecting to std::chrono then let me be clear I have absolutely no issue with that.

Concerning std::string I also understand your point of view, but that decision was made some time ago, based on actual experience with string things getting tangled up. It's also good to remember that  many people are not so well-versed in the C++ standard library as you are. :)  
Overall, I think it's just one of those compromises of using the C++ _language_ in a very Arduino _library_ context.

> For now, let's not squabble in front of the children. :-)

Didn't mean to. I wanted to avoid a situation where @mggates39 would spend time on `str::string`ifying things, and me then going "nice, but no".

Reply:
Yeah, I came across rough here.  I agree that the std::chrono is the way to go.  I had just never heard of it and the existing code was not using it.  Definitely taking this back to the drawing board.